packages/sanity/src/core/releases/tool/detail/useBundleDocumentsValidation.ts

```diff
@@ -0,0 +1,38 @@
+import {type SanityDocument} from '@sanity/types'
+import {useMemo} from 'react'
+import {useObservable} from 'react-rx'
+import {useDocumentPreviewStore, useSchema, useSource} from 'sanity'
+
+import {bundleDocumentsValidation, type DocumentValidationStatus} from './bundleDocumentsValidation'
+
+export function useBundleDocumentsValidation(bundles: SanityDocument[]) {
+  const {getClient, i18n} = useSource()
+  const {unstable_observeDocumentPairAvailability, unstable_observeDocument} =
+    useDocumentPreviewStore()
+  const schema = useSchema()
+  // This will be stable across renders, will emit a new one when a bundle is added or removed.
+  const bundlesIds = bundles.map((bundle) => bundle._id).join(',')
+
+  const observable = useMemo(() => {
+    return bundleDocumentsValidation(
+      {
+        observeDocumentPairAvailability: unstable_observeDocumentPairAvailability,
+        observeDocument: unstable_observeDocument,
+        schema,
+        i18n,
+        getClient,
+      },
+      bundlesIds.split(','),
```

Comment:
Should make a note here that this is a hack and something that we should look into fixing later